cirq-core/cirq/sim/act_on_state_vector_args.py

```diff
@@ -39,7 +39,8 @@ class ActOnStateVectorArgs(ActOnArgs):
 
     def __init__(
         self,
-        target_tensor: np.ndarray,
+        target_tensor: Union[np.ndarray, 'cirq.STATE_VECTOR_LIKE'] = 0,
+        dtype: Type[np.number] = np.complex64,
```

Comment:
New param `dtype` needs to be moved to the bottom.

Reply:
Sure.